swag-2024-Q1-openday/slides.md

```diff
@@ -0,0 +1,207 @@
+- Forward Error Correction (FEC): Add redundant information to the transmitted packets, Allowing receiver to reconstruct lost packets even if they are not received.
+
+- NACK: NACK notify sender to retransmit missing packets and fill in the gaps to maintain playback continuity.
+
+- Adaptive bitrate control: Dynamically adjust the bitrate of the transmitted media stream based on network conditions.
+
+---
+![original](background.png)
+
+# WebRTC - Summary
+
+- Pros:
+    - Strong security ensured by SRTP[^2]
+    - Strong Community support, work on almost every browser.
+    - High stability under bad network
+- Cons:
+    - Hard to scale when there's multi participants
+    
+
+[^2]: [Secure Real-Time Transport Protocol](https://datatracker.ietf.org/doc/html/rfc3711)
+
+---
+![original](background.png)
+
+# SRT
+
+- **UDP** based
+- Supported codec: codec agnostic
+- Lattency: (< 500ms)
+- Loss packet handling: FEC, ARQ, Too-late packet
+- Timestamp-Based Packet Delivery (TSBPD)
+
+---
+[.build-lists: true]
+
+![original](background.png)
+
+# Why SRT codec agnostic ?
+- It doesn't have any codec limitation on data processing, 
+the responsibility of encoding/decoding fall on upstream app using SRT.
+- Acting as a normal udp packet wrap SRT content.
+
+---
+![original](background.png)
+
+# SRT - Summary
+
+- Pros:
+    - Codec agnostic
+    - Strong security ensured by AES encrypted content.
+    - High stability under bad network
+- Cons:
+    - Not support by native web, require special player.
+    - Require extra bandwidth for SRT machanism maintain stream quality
+
+---
+![original](background.png)
+
+| Protocol | RTMP | WebRTC | SRT |
+| --- | --- | --- |--- |
+| Supported codecs |H.264, AAC| H.264, VP9, VP8, Opus, G.711 G.722, iLBC, iSAC | Unlimited |
+| Latency | < 5s | < 500ms | < 500ms|
+| Security | Need Extension | Built in | Built in (AES) |
+| Disruption Tolerance | Average | Good | Good |
+
+---
+[.build-lists: true]
+<!-- ![original](background.png) -->
+# [fit] Quest: What video/audio codec does 
+# [fit] **RTMP** support ?
+
+- video: H.264
+- audio: AAC
+
+---
+<!-- ![original](background.png) -->
+
+# [fit] Deep dive into
+# [fit]  **SRT**
+
+---
+![original](background.png)
+
+# Timestamp-Based Packet Delivery
+Timestamps allow the decoder to **reorder** packets if they arrive out of order due to network delays or retransmissions. Ensuring frames are decoded and displayed in the correct order.
+
+---
+![inline](srt-tsbpd.png)
+
+--- 
+![original](background.png)
+
+# Automatic Repeat Request
+
+---
+# Packet Delivery Time
+[TODO]
+- Describe fomulla.
+
+---
+# [fit] Quest: packet delivery time 
+
+---
+![original](background.png)
+# Too Late Packet
+
+---
+![inline](srt-too-late-packet-drop.png)
+
+
+---
+![original](background.png)
+
+# WebRTC References:
+- https://medium.com/agora-io/how-does-webrtc-work-996748603141
+- https://www.100ms.live/blog/webrtc-turn-server
+
+---
+![original](background.png)
```

Comment:
Why didn't we use WebRTC back then ?
- It wasn't mature enough
- Develope time constraints (RTMP is better supported all around)